### PR TITLE
Bug 1690007: Add the ability to create labels during Deploy Image creation

### PR DESCRIPTION
**Pull request by rhamilto**
Resolves https://jira.coreos.com/browse/CONSOLE-833
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1690007

Screen Shot 2019-04-25 at 9 45 25 AM

### CODE REVIEW

**Comment by rhamilto** [comment]
/hold

Found a bug

**Comment by rhamilto** [comment]
/hold cancel

Updated to remove magic updating of default label (which was the source of the bug and a usability concern).

File: frontend/public/components/deploy-image.tsx [hunk 1]
```diff
@@ -433,6 +445,7 @@ export type DeployImageState = {
   loading: boolean,
   isi?: any,
   name: string,
+  labels: any,
```
Comment (spadgett) [inline frontend/public/components/deploy-image.tsx]: We should change `,` to `;` in the type declaration

```suggestion
  labels: string[];
```

File: frontend/public/components/deploy-image.tsx [hunk 2]
```diff
@@ -90,6 +92,10 @@ export class DeployImage extends React.Component<DeployImageProps, DeployImageSt
     this.setState({name: event.currentTarget.value});
   };
 
+  onLabelsChange: React.ReactEventHandler<HTMLInputElement> = labels => {
```
Comment (spadgett) [inline frontend/public/components/deploy-image.tsx]: This looks like the wrong type if the method doesn't take an event.

```suggestion
  onLabelsChange = (labels: string[]) => {
```

**Comment by rhamilto** [comment]
Comments addressed.  Thanks, @spadgett.

**Comment by rhamilto** [comment]
Aligned label behavior with #1493.

File: frontend/public/components/deploy-image.tsx [hunk 3]
```diff
@@ -397,6 +403,13 @@ export class DeployImage extends React.Component<DeployImageProps, DeployImageSt
                   <div className="help-block">Identifies the resources created for this image.</div>
                 </div>
               </div>
+              <div className="form-group co-deploy-image__labels">
+                <label htmlFor="tags-input" className="control-label">Labels</label>
+                <SelectorInput labelClassName="co-text-deploymentconfig" onChange={this.onLabelsChange} tags={labels} />
+                <div className="help-block" id="labels-help">
+                  {`If no labels are specified, app=${name || '<name>'} will be added.`}
```
Comment (spadgett) [inline frontend/public/components/deploy-image.tsx]: Missed this in the other PR, but...

```suggestion
                  If no labels are specified, app={name || '<name>'} will be added.
```

Comment (rhamilto) [reply frontend/public/components/deploy-image.tsx]: Thanks.  Fixed.

**Pull request merged**